articles/model-anatomy.adoc

```diff
@@ -0,0 +1,644 @@
+In our system, a `person` without known `name` might be a problem, but we could still process them without knowing their `age`.
+
+A property's type describes what kind of values can appear in that property.
+All LionWeb data types, and thus all properties, are _value typesfootnote:[https://en.wikipedia.org/w/index.php?title=Value_type]_.
+This means they don't have any identity, we just know their value: We can distinguish `10` and `15`, but `12` and `12` are the same.
+
+Some modeling frameworks allow properties with multiple values, e.g. `luckyNumbers: Integer[0..*] = {23, 42, 4711}`.
+LionWeb does not support that, because the values don't have identity.
+So if we first had a property value of `{23, 42}`, and then `{23, 42, 42}`, we have no way of telling whether the new `42` has been inserted or appended.
+This leads to all kinds of problems we'd like to avoid.
+
+The following subsections describe the kinds of data types supported by LionWeb.
+
+==== Primitive types
+LionWeb ships with built-in types for `String`, `Integer`, and `Boolean`, as they are needed in almost all systems.
+String properties can have any string value, including empty.
+Integer properties support positive and negative decimal integers of any length.
+Boolean properties can only have the values `true` or `false`.
+A language might define its own additional _primitive types_.
+In any case, we store all of them as string on low-level serialization format (i.e. JSON).
+
+==== Enumerations
+A property can also be of an _enumeration_ type.
+Enumerations have a _name_, _key_, and a list of _enumeration literals_.
+Again, name is for human understanding, and key for unambiguous technical reference.
+Each literal also has a _name_ and a _key_, with the same purpose.
+A language must explicitly declare an enumeration to use it as the type of a property.
+Enumeration properties must have one of the enumeration literal values.
+On low-level serialization, we store the value as the enumeration literal's key.
+
+[plantuml, personEnum, svg]
+----
+hide empty members
+
+enum status as "CivilStatus [key=cs]"
+{
+  single [key=s]
+  married [key=m]
+  divorced [key=d]
+  widowed [key=w]
+}
+----
+
+==== Structured data types
+LionWeb supports another kind of simple type: _structured data type_ (added in LionWeb version 2024.1).
+They are a collection of other primitive types, called _fields_.
+Both the structured data type and each field has a _name_ and _key_ with the usual meaning.
+
+[plantuml, sdtColor, svg]
+----
+json "TextPosition [key=tp]" as tp {
+  "line [key=lin]": "Integer",
+  "column [key=col]": "Integer"
+}
+
+json "TextRange [key=tr]" as tr {
+  "start [key=st]": "TextPosition",
+  "end [key=en]": "TextPosition",
+  "label [key=la]": "String",
+  "valid [key=va]": "Boolean"
+}
+----
+
+We store the value of a structured data type as string, the same as all other primitive types.
+The content of that string is an escaped JSON structure.
```

Comment:
"is a JSON structure serialized as a JSON string" would be more accurate, I think.

Reply:
I tried quite a bit to describe this accurately here, but didn't succeed. That's why this is the only place where I refer to the spec.

What about "is a JSON object as escaped string"?